rotkehlchen/history/events/structures/asset_movement.py

```diff
@@ -0,0 +1,208 @@
+        `unique_id`: Unique identifier for this asset movement.
+            Either the exchange transaction id or the associated onchain transaction hash.
+            Used in conjunction with location to generate the event identifier.
+        `is_fee`: Controls whether this represents a fee event.
+        """
+        location_name = get_formatted_location_name(location)
+        asset_symbol = asset.resolve_to_asset_with_symbol().symbol
+        if is_fee:
+            sequence_index = 1
+            event_subtype = HistoryEventSubType.FEE
+            notes = f'Pay {balance.amount} {asset_symbol} as {location_name} {str(event_type).lower()} fee'  # noqa: E501
+        else:
+            sequence_index = 0
+            if event_type == HistoryEventType.DEPOSIT:
+                event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                notes = f'Deposit {balance.amount} {asset_symbol} to {location_name}'
+            else:
+                event_subtype = HistoryEventSubType.REMOVE_ASSET
+                notes = f'Withdraw {balance.amount} {asset_symbol} from {location_name}'
+
+        super().__init__(
+            event_identifier=event_identifier if event_identifier is not None else self._create_event_identifier(  # noqa: E501
+                location=location,
+                timestamp=timestamp,
+                asset=asset,
+                balance=balance,
+                unique_id=unique_id,
+            ),
+            sequence_index=sequence_index,
+            timestamp=timestamp,
+            location=location,
+            event_type=event_type,
+            event_subtype=event_subtype,
+            asset=asset,
+            balance=balance,
+            notes=notes,
+            identifier=identifier,
+            extra_data=extra_data,
+        )
+
+    @staticmethod
+    def _create_event_identifier(
+            location: Location,
+            timestamp: TimestampMS,
+            asset: Asset,
+            balance: Balance,
+            unique_id: str | None,
+    ) -> str:
+        """Create a unique event identifier from the given parameters.
+        `unique_id` is a transaction id from the exchange, which in combination with the
+        location makes a unique event identifier. If this is not available, the location,
+        timestamp, asset, and balance must all be combined to ensure a unique identifier.
+        """
+        if unique_id is not None:
+            return hash_id(str(location) + unique_id)
+
+        return hash_id(
+            str(location) +
+            str(timestamp) +
+            asset.identifier +
+            str(balance.amount),
+        )
+
+    @property
+    def entry_type(self) -> HistoryBaseEntryType:
+        return HistoryBaseEntryType.ASSET_MOVEMENT_EVENT
+
+    def serialize_for_db(self) -> tuple[tuple[str, str, HISTORY_EVENT_DB_TUPLE_WRITE]]:
+        return (self._serialize_base_tuple_for_db(),)
+
+    @staticmethod
+    def _check_deserialized_event_type(
```

Comment:
So don't have this into a separate function. You don't need this in many places.

1. Deserializing from the outside, so `deserialize` yes you would need to check. But just add it as a simple if check.
2. Deserializing from the DB you should not need to do that as the DB data have already been checked at addition. We don't recheck DB data when reading it out of the DB.